candidate/models.py

```diff
@@ -1452,6 +1460,20 @@ def update_candidate_row_entry(self, candidate_we_vote_id, update_values):
                 if 'facebook_url' in update_values:
                     existing_candidate_entry.facebook_url = update_values['facebook_url']
                     values_changed = True
+                if 'photo_url' in update_values:
+                    # check if candidate has an existing photo in the CandidateCampaign table
+                    if positive_value_exists(existing_candidate_entry.we_vote_hosted_profile_image_url_large) and \
+                            positive_value_exists(existing_candidate_entry.we_vote_hosted_profile_image_url_medium) \
+                            and positive_value_exists(existing_candidate_entry.we_vote_hosted_profile_image_url_tiny):
+                        save_to_candidate_object = False
+                    else:
+                        save_to_candidate_object = True
+
+                    candidate_results = self.modify_candidate_with_organization_endorsements_image(
+                        existing_candidate_entry, update_values['photo_url'], save_to_candidate_object)
+                    if candidate_results['success']:
+                        values_changed = True
+
                 # now go ahead and save this entry (update)
```

Comment:
I think you might want to get the updated "existing_candidate_entry" from the "candidate_results" here before the save a few lines down.